lndr-backend/src/Lndr/Handler/Identity.hs

```diff
@@ -0,0 +1,77 @@
+{-# LANGUAGE OverloadedStrings #-}
+{-# LANGUAGE TupleSections #-}
+
+module Lndr.Handler.Identity where
+
+import           Control.Concurrent.STM
+import           Control.Lens
+import           Control.Monad.Except
+import           Control.Monad.Reader
+import           Control.Monad.Trans.Resource
+import qualified Data.ByteString.Base64  as B64
+import qualified Data.ByteString.Lazy    as LBS
+import           Data.Pool               (withResource)
+import           Data.Text               (Text)
+import qualified Data.Text               as T
+import qualified Data.Text.Encoding      as T
+import qualified Lndr.Db                 as Db
+import           Lndr.Handler.Types
+import           Lndr.IdentityVerification
+import           Lndr.Signature
+import           Lndr.Types              as Types
+import           Lndr.Util
+import           Network.Ethereum.Web3
+import           Network.HTTP.Client
+import           Network.HTTP.Client.MultipartFormData
+import           Servant
+import           Text.EmailAddress
+import           System.Log.FastLogger
+
+verifyIdentityHandler :: IdentityVerificationRequest -> LndrHandler NoContent
+verifyIdentityHandler req@(IdentityVerificationRequest email addr info requiredDocs signature) = do
+  unless (Right addr == recoverSigner req) $ throwError (err401 {errBody = "Bad signature."})
+  (ServerState pool configTVar loggerSet) <- ask
+  config <- liftIO $ readTVarIO configTVar
+  response <- liftIO $ sendVerificationRequest config req
+
+  liftIO . mapM (\doc -> sendVerificationDocument loggerSet config (Types.id response) doc) $ idDocs info
```

Comment:
Should `idDocs` here be `requiredDocs`?

Reply:
No, what's happening here is that after the user is created, we send the id documents one at a time to the SumSub API. The `requiredIdDocs` list is part of the original account creation, and then we send the attached documents one at a time. The idea is to make it simpler for the user, although we could have the user create a KYC account and upload documents separately

Reply:
Ah, I see. `idDocs` is a property of `info`. Are there any parameters in this function that can be `_`-ed out? Looks like `signature`, `requiredDocs`, and `email` are not being used here.